src/app/list-manager/list-item/list-item.component.html

```diff
@@ -0,0 +1,197 @@
+
+    </div>
+    <div class="d-flex justify-content-center align-items-center mb-3">
+        <div class="text-truncate fs-8 pointer"
+            [title]="i18n('notepad2.note.lastModified', [fdate(clonedListItem.updated , 'Pp')])">
+            {{ i18n('notepad2.note.lastModified', [fdist(clonedListItem.updated, true)]) }}
+        </div>
+    </div>
+    }
+</form>
+
+<ng-template #genericListBody>
+    @for (item of clonedListItem.items; track $index) {
+    <!-- Display all unchecked items first (or all items if list is configured this way) -->
+    @if (item.checked === false || this.editableList.controls.checkedBelow.value !== true) {
+    <ng-container *ngTemplateOutlet="genericListItem; context: { item: item, i: $index }"></ng-container>
+    }
+    <!-- Display all unchecked items first -->
+    }
+    @for (item of clonedListItem.items; track $index) {
+    <!-- Display checked items second -->
+    @if (item.checked === true && this.editableList.controls.checkedBelow.value === true) {
+    <ng-container *ngTemplateOutlet="genericListItem; context: { item: item, i: $index }"></ng-container>
+    }
+    <!-- Display checked items second -->
+    }
+</ng-template>
+
+<ng-template #genericListItem let-item="item" let-i="i">
+    <li class="list-group-item d-flex align-items-center py-1 px-2">
+        @if (list.listStyle === 'ul') {
+        <div>
+            <span class="me-2">•</span>
+            <span contenteditable="true">A list item</span>
+        </div>
+        }
+        @else if (list.listStyle === 'cb') {
+        <div class="flex-fill d-flex align-items-center">
+            <input class="form-check-input me-2 mt-0" type="checkbox" [(ngModel)]="item.checked"
+                (change)="onCheckListItem(i)" [name]="'list-item-' + i + '-checkbox'"
+                [id]="'list-item-' + i + '-checkbox'">
+            <!-- <input class="form-control form-control-sm plain flex-fill me-3" type="text" [(ngModel)]="item.content"
+                (keyup)="onKeyUp()" [name]="'list-item-' + $index + '-content'"
+                [id]="'list-item-' + $index + '-content'" autocomplete="off" [autofocus]="$index == 0"> -->
+            <p [(contentEditableWithBinding)]="item.content" class="flex-fill me-3 mb-0 editable-item"
+                [id]="'list-item-' + i + '-content'" autocomplete="off" [autofocus]="i == 0" [attr.data-note-index]="i"
+                (keydown)="onItemTextKeydown(i, $event)" (keyup)="onKeyUp()"></p>
+        </div>
+        }
+        @else {
+        <div class="ms-2" contenteditable="true">
+            A list item
+        </div>
+        }
+        <div class="ms-auto">
+            <span class="fs-8 fst-italic mx-2">{{i + 1}}</span>
+            <button type="button" class="btn btn-sm btn-outline-danger" (click)="onDeleteListItem(i)">
+                <app-icon [iconClass]="icons['delete']"></app-icon>
+            </button>
+        </div>
+    </li>
+</ng-template>
+}
```

Comment:
Reusable components are very hard to implement in this case. This is because there're multiple ways to manipulate the list items by the user. Handling all possibilities with a good user experience have been realized with this single component. So I won't change this.